library/pkparse.c

```diff
@@ -851,6 +851,15 @@ int mbedtls_pk_parse_subpubkey(unsigned char **p, const unsigned char *end,
 #if defined(MBEDTLS_RSA_C)
     if (pk_alg == MBEDTLS_PK_RSA) {
         ret = mbedtls_rsa_parse_pubkey(mbedtls_pk_rsa(*pk), *p, (size_t) (end - *p));
+        if (ret == 0) {
+            /* On success all the input has been consumed by the parsing function. */
+            *p += end - *p;
+        } else if (ret & 0x7f) {
```

Comment:
Is this the right condition? Before adding high-level code we should ensure that have _only_ a low-level code, which I think would be `(ret & 0x7f) == ret`. Am I missing something?

Reply:
ops, correct  ':)